test/test_linalg.py

```diff
@@ -7449,6 +7449,42 @@ def test_permute_matmul(self):
         c = a.permute(0, 1, 3, 2).matmul(b)
         self.assertEqual([c.min(), c.max(), c.sum()], [24, 24, 414720])
 
+    def test_bfloat16_accumulation_with_ref_path(self):
+        # fix https://github.com/pytorch/pytorch/issues/95125
+        # and https://github.com/pytorch/pytorch/issues/83863
+        # for bf16 accumulation in gemm ref path
+        def check_correctness(fn, *args):
+                expected = fn(*args).bfloat16()
+                with torch.backends.mkldnn.flags(enabled=False):
+                    def test():
+                        bf16_args = (arg.bfloat16() for arg in args)
+                        tmp_result=fn(*bf16_args)
+                        return tmp_result
+                    c = test()
+                    assert(torch.all(c == expected)), "Incorrect result with\n" \
+                                                      f"expected: {expected}\n" \
+                                                      f"got: {c}\n"
+        # test matmul
+        a = torch.ones(300, 300)
```

Comment:
need to add cases when a or b is not contiguous here.

Reply:
Thanks for reminder, have added checks for `notrans`, `transa`, `transb`, and `transab` for different `gemm_core_` impl